pkg/cloud/project.go

```diff
@@ -43,3 +43,10 @@ type SingleProjectRouter struct {
 func (r *SingleProjectRouter) ProjectID(ctx context.Context, version meta.Version, service string) string {
 	return r.ID
 }
+
+func getProjectID(ctx context.Context, pr ProjectRouter, opt Option, version meta.Version, service string) string {
```

Comment:
This is a helper function to avoid having to put a lot of logic into the generate code (it is easier to maintain).

Reply:
This should be opt-in no? The default is that projectID is not set, so this should have no effect and be taken directly from the router.

Reply:
I suggested opt-in , because 'm assuming before this PR it goes to the router

Reply:
I might not be understanding your comment.

The reason this is pulled out into a func is to make the change smaller in the generated code, which is harder to maintain.

Now, the behavior is opt-in:

- By default, the allOptions.projectID will be empty string ("") which means we go to the ProjectRouter to figure out which project we should use.
- Only if you add the ForceProjectID() option will we end up with an override.

I think this is sufficient for an opt-in. We could add a global flag, but it seems excessive -- it would be an additional layer of complexity for what seems like not enough gain.